File: src/main/java/com/procurement/orchestrator/domain/dto/auction/ElectronicAuctions.java
package com.procurement.orchestrator.domain.dto.auction;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

import java.util.Set;

@Data
@JsonInclude(JsonInclude.Include.NON_NULL)
public class ElectronicAuctions {

    @JsonProperty(value = "details")
    private Set<ElectronicAuctionsDetails> details;

    public ElectronicAuctions(@JsonProperty("details") final Set<ElectronicAuctionsDetails> details) {
        this.details = details;
    }
}

File: src/main/kotlin/com/procurement/orchestrator/domain/model/tender/auction/ElectronicAuctions.kt
package com.procurement.orchestrator.domain.model.tender.auction

import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.annotation.JsonProperty
import com.procurement.orchestrator.domain.model.ComplexObject
import java.io.Serializable

data class ElectronicAuctions(
    @field:JsonInclude(JsonInclude.Include.NON_EMPTY)
    @field:JsonProperty("details") @param:JsonProperty("details") val details: ElectronicAuctionsDetails = ElectronicAuctionsDetails()
) : ComplexObject<ElectronicAuctions>, Serializable {

    override fun updateBy(src: ElectronicAuctions) = ElectronicAuctions(
        details = details updateBy src.details
    )
}
